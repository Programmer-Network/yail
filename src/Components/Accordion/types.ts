import { IDraggableListItem } from "../DraggableList/types";

export interface IAccordionSectionItem {
  id: number;
  title: string;
  order: number;
}

export interface ISection {
  id: number;
  description?: string;
  title: string;
  order: number;
  items: IAccordionSectionItem[];
}

export enum AccordionOrderType {
  SECTIONS = "sections",
  LECTURES = "lectures"
}

export interface IAccordionProps {
  className?: string;
  sections: ISection[];
  sectionTitleClassName?: string;
  onSorted?: (
    type: AccordionOrderType,
    sections: ISection[],
    section: { sectionId: number; items: IDraggableListItem[] }
  ) => void;
  onSectionItemClick?: (item: IDraggableListItem) => void;
  onSectionClick?: (item: IDraggableListItem) => void;
  onSelected?: (item: IDraggableListItem) => void;
  onAddSection?: () => void;
  addSectionItemLabel?: string;
  onAddSectionItem?: (item: ISection) => void;
  addSectionLabel?: string;
  expanded: number[];
  setExpanded: (sections: number[]) => void;
  hasDraggableSections?: boolean;
  hasDraggableSectionItems?: boolean;
  defaultExpanded?: boolean;
  selected?: {
    sectionId: number | null;
    itemId: number | null;
  };
  setSelectedSection?: (sectionId: number | null) => void;
  setSelectedItem?: (itemId: number | null) => void;
}
