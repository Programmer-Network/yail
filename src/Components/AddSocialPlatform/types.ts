export interface IAddSocialPlatformProps {
  className?: string;
  platforms: { name: string; url: string }[];
  onAdd: (data: Record<string, string>) => void;
}
