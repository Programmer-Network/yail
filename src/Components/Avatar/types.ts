export interface IAvatarProps {
  src?: string;
  alt?: string;
  size?: number;
  colors?: string[];
  className?: string;
}
