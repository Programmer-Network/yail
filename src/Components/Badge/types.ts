export enum BadgeVariantEnum {
  PRIMARY = "primary",
  SECONDARY = "secondary",
  SUCCESS = "success",
  WARNING = "warning",
  ERROR = "error",
  BETA = "beta",
  NEW = "new"
}

export interface IBadgeProps {
  title: string;
  className?: string;
  variant: BadgeVariantEnum;
}
