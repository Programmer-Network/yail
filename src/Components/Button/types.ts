import { ReactNode } from "react";

import { IconName } from "Components/Icons/types";

export enum ButtonVariantEnum {
  PRIMARY = "primary",
  SECONDARY = "secondary"
}

export interface IButtonProps {
  onClick?: (e: React.MouseEvent) => void;
  children?: ReactNode;
  className?: string;
  disabled?: boolean;
  type?: "button" | "submit" | "reset";
  isLoading?: boolean;
  outlined?: boolean;
  icon?: {
    iconName?: IconName;
    iconClassName?: string;
    iconPosition?: "left" | "right";
  };
  variant?: ButtonVariantEnum;
}
