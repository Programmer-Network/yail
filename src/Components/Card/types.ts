import { NavLinkProps } from "react-router-dom";

export interface ICardData {
  title: string;
  titleUrl: string;
  description: string;
  date?: string;
  author?: string;
  authorUrl?: string;
  avatarUrl?: string;
  tags?: {
    name: string;
    url: string;
  }[];
  image?: string;
}

export interface ICard {
  data: ICardData;
  NavLink: React.ForwardRefExoticComponent<
    NavLinkProps & React.RefAttributes<HTMLAnchorElement>
  >;
  className?: string;
}
