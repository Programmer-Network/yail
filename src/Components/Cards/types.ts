import { NavLinkProps } from "react-router-dom";

import { ICardData } from "../Card/types";

export interface ICardsProps {
  cards: ICardData[];
  NavLink: React.ForwardRefExoticComponent<
    NavLinkProps & React.RefAttributes<HTMLAnchorElement>
  >;
  columns: number;
  className?: string;
  cardClassName?: string;
}
