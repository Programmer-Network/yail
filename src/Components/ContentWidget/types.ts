export interface IContentWidgetBaseItem {
  id?: number;
  title: string;
}

export interface IContentWidgetItemMore {
  tooltipText: string;
  iconName?: string;
  iconClassName?: string;
  onClick: () => void;
}

export interface IContentWidgetItem<T extends IContentWidgetBaseItem> {
  item: T;
  onClick: (item: T) => void;
}

export interface IContentWidgetProps<T extends IContentWidgetBaseItem> {
  title: string;
  className?: string;
  isLoading: boolean;
  items: IContentWidgetItem<T>[];
  more?: IContentWidgetItemMore;
}
