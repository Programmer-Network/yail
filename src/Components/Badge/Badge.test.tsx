import { render, screen } from "@testing-library/react";

import { Badge } from "./";
import { BadgeVariantEnum } from "./types";

describe("Badge component", () => {
  const testTitle = "Test Badge";

  test("renders correctly - snapshot test", () => {
    const { asFragment } = render(
      <Badge title={testTitle} variant={BadgeVariantEnum.FILLED} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders the title correctly", () => {
    render(<Badge title={testTitle} variant={BadgeVariantEnum.FILLED} />);
    const badgeElement = screen.getByText(testTitle);
    expect(badgeElement).toBeInTheDocument();
  });

  test("applies filled variant", () => {
    render(<Badge title={testTitle} variant={BadgeVariantEnum.FILLED} />);
    const badgeElement = screen.getByText(testTitle);
    expect(badgeElement).toHaveClass("bg-primary");
  });

  test("applies custom class name", () => {
    const customClass = "custom-class";
    render(
      <Badge
        title={testTitle}
        variant={BadgeVariantEnum.FILLED}
        className={customClass}
      />
    );
    const badgeElement = screen.getByText(testTitle);
    expect(badgeElement).toHaveClass(customClass);
  });

  test("applies outline variant", () => {
    render(<Badge title={testTitle} variant={BadgeVariantEnum.OUTLINE} />);
    const badgeElement = screen.getByText(testTitle);
    expect(badgeElement).toHaveClass("bg-transparent");
  });

  test("applies clickable class name", () => {
    render(
      <Badge
        title={testTitle}
        variant={BadgeVariantEnum.FILLED}
        onClick={() => alert("Badge clicked")}
      />
    );

    const badgeElement = screen.getByText(testTitle);
    expect(badgeElement).toHaveClass("cursor-pointer");
  });
});
