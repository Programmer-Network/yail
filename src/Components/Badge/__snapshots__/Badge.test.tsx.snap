// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge component > renders correctly - snapshot test 1`] = `
<DocumentFragment>
  <span
    class="yl:rounded yl:border yl:border-border yl:px-1 yl:pb-[2px] yl:pt-[1px] yl:text-[10px] yl:text-text"
  >
    Test Badge
  </span>
</DocumentFragment>
`;
